snap/info.go

```diff
@@ -203,7 +203,7 @@ func snapDataDir(opts *dirs.SnapDirOptions) string {
 		opts = &dirs.SnapDirOptions{}
 	}
 
-	if opts.HiddenSnapDataDir {
+	if opts.HideSnapDir && opts.MigratedHidden {
```

Comment:
then we wouldn't need to touch this I think

Reply:
Sorry, this change is actually wrong. It should be:

```suggestion
	if opts.MigratedHidden {
```

I changed the semantics of the opts after writing this and forgot to adjust these cases. The if case is then the same, the field was just renamed (maybe it should've remained the same and the other field be named something else).

I'm not sure having a separate struct for the state related parts of the hidden opts will improve much. We'd need to have code to map the state opts into the dir opts in a lot of places. Besides that, sometimes it's also necessary to make artificial opts so we get one of the directories specifically (e.g., when doing clean ups, searching, backend ops, we something want the hidden/expose dir specifically), which breaks the separation between the two layers. Of course the code can just make a state opts that it knows would be mapped into the right SnapDirOptions, but then there's no abstraction and the mapping logic isn't encapsulated in a particular function, it's spread all over. Another issue is import cycles, since the new options need snapstate info to be created and they'll be used all over. Do you have a concrete idea on how it should look?

Reply:
sorry, I'm really confused, why would most places outside of snapstate care about potential vs which directory they should actually use?

> I'm not sure having a separate struct for the state related parts of the hidden opts will improve much

I find the new fields of the structs quitse confusing, they make visible to everything aspects  that seems implementation details of the migration, why for example should the snap package know about anything but which directories to pick?

Reply:
> Do you have a concrete idea on how it should look?

afaict .HideSnapDir is used/should be only inside snapstate. So I would the SnapDirOptions definition to be reverted. And snapstate to use an internal helper below GetSnapDirOptions for its own full internal state. I'm not sure a 2nd struct is needed maybe not, this other helper could just return a bool, not sure.

Reply:
Hm, I'm probably a bit confused. I'll play around with the code and make some changes so I understand this better. If I have any doubts, I'll reach out to you tomorrow